app/services/puppy_aging_service.rb

```diff
@@ -0,0 +1,36 @@
+class PuppyAgingService
+  def initialize(puppy)
+    @puppy = puppy
+  end
+
+  def process
+    time_elapsed = change_time
+    update_puppy(time_elapsed)
+    @puppy
```

Comment:
great implicit return, after the update passing the puppy out to the controller.